### ISSUE TITLE
netstandard*: MissingMethodException due to "bad" `System.IO.FileSystem.Primitives.dll` from nuget

### ISSUE THREAD

**Issue by radical** [mono/mono#12692]
The project (`netstandard2.0`) references `<PackageReference Include="Mono.Cecil" Version="0.10.2" />` and copies the assemblies from nuget to the output dir.

Test code:
```
using Mono.Cecil;
class Test{
        static void Trigger() => ModuleDefinition.ReadModule("/tmp/q", new ReaderParameters());
        public static void Main() => Trigger();
}
```

This throws:
```
Unhandled Exception:
System.MissingMethodException: Method not found: void System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
  at Mono.Cecil.ModuleDefinition.ReadModule (System.String fileName, Mono.Cecil.ReaderParameters parameters) [0x00008] in <702d0f72947148848a6f81a1a562dbae>:0
  at Test.Trigger () [0x0000a] in <00585c53a9ef4234ab8b692a53c9a46b>:0
  at Test.Main () [0x00000] in <00585c53a9ef4234ab8b692a53c9a46b>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.MissingMethodException: Method not found: void System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
  at Mono.Cecil.ModuleDefinition.ReadModule (System.String fileName, Mono.Cecil.ReaderParameters parameters) [0x00008] in <702d0f72947148848a6f81a1a562dbae>:0
  at Test.Trigger () [0x0000a] in <00585c53a9ef4234ab8b692a53c9a46b>:0
  at Test.Main () [0x00000] in <00585c53a9ef4234ab8b692a53c9a46b>:0
```

The problem seems to be `System.IO.FileSystem.Primitives.dll` that gets copied to the output dir. If you replace that with the one from mono (`lib/mono/4.5/Facades`) then it works (and throws about the missing file `/tmp/q`).

Running with `MONO_LOG_LEVEL=debug MONO_LOG_MASK=asm`, I get:
```
Mono: Loading reference 10 of /Users/radical/tmp/t1/bin/Debug/netstandard2.0/Mono.Cecil.dll asmctx DEFAULT, looking for System.IO.FileSystem.Primitives, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
Mono: Assembly System.IO.FileSystem.Primitives is a framework Facade asseembly
Mono: Assembly System.IO.FileSystem.Primitives is a framework Facade asseembly
Mono: Domain foo.dll search path is:
Mono:   path[0] = '/Users/radical/tmp/t1/bin/Debug/netstandard2.0/'
Mono: End of domain foo.dll search path.
Mono: Assembly Loader probing location: '/Users/radical/tmp/t1/bin/Debug/netstandard2.0/System.IO.FileSystem.Primitives.dll'.
Mono: Image addref System.IO.FileSystem.Primitives[0x7fb847d3ded0] (asmctx DEFAULT) -> /Users/radical/tmp/t1/bin/Debug/netstandard2.0/System.IO.FileSystem.Primitives.dll[0x7fb848856a00]: 2
Mono: Prepared to set up assembly 'System.IO.FileSystem.Primitives' (/Users/radical/tmp/t1/bin/Debug/netstandard2.0/System.IO.FileSystem.Primitives.dll)
Mono: Assembly System.IO.FileSystem.Primitives[0x7fb847d3ded0] added to domain foo.dll, ref_count=1
Mono: Assembly Loader loaded assembly from location: '/Users/radical/tmp/t1/bin/Debug/netstandard2.0/System.IO.FileSystem.Primitives.dll'.
Mono: Assembly Ref addref Mono.Cecil[0x7fb847d2f350] -> System.IO.FileSystem.Primitives[0x7fb847d3ded0]: 2
```

## On which platforms did you notice this

[ X ] macOS
[ ] Linux
[ ] Windows

**Version Used**:

mono:

`Mono JIT compiler version 5.22.0.14 (2018-12/33840d42108 Fri Dec 21 12:00:22 PST 2018)`

**Comment by radical** [comment]
Test project:
[test-missing-method.zip](https://github.com/mono/mono/files/2815050/test-missing-method.zip)

**Comment by lambdageek** [comment]
Doesn't work with 5.4.1.7 either.  Not a regression.

Also this isn't due to a new (0.10.2) Mono.Cecil - it doesn't work with 0.10.1.

Did this ever work?

Wouldn't pretty much all nestandard1.x-era corefx assemblies need to be on the denied assembly list?  

For example our 4.5/Facades/System.Collections.NonGeneric.dll is full of forwarders, but `.nuget/packages/system.collections.nongeneric/4.0.1/lib/netstandard1.3/System.Collections.NonGeneric.dll` is full of code.   So if I have some package X built targeting nestandard1.3 that uses `System.Collections.ArrayList` and then I use X from some other package that has `<CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>` would I then end up with a bad `System.Collections.NonGeneric.dll` in my `bin/Debug/netstandard2.0` directory?

It's possible I'm pretty confused about how this should work.

**Comment by marek-safar** [comment]
I'd consider this to be packaging issue on Cecil side. To be fully compatible with NS2.0 it should be using System.IO.FileSystem.Primitives 4.3.0 version or newer and not 4.0.1

**Comment by chkn** [comment]
@marek-safar I thought NS2.0 should be compatible with earlier versions of .NET Standard? I believe Cecil is packaged as NS1.3 ( https://www.fuget.org/packages/Mono.Cecil )

**Comment by marek-safar** [comment]
That's right but this is a bug in System.IO.FileSystem.Primitives where version 4.0.1 has wrong setup for .NET 4.6 TFM which is fixed in 4.3.0. I think Cecil can still target NS1.3 but with updated System.IO.FileSystem.Primitives dependency.
